AsyncMux/Sources/MuxDB.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  MuxDB.swift
+//  AsyncMux
+//
+//  Created by Lissy Harrison on 6/5/23.
+//
+
+import Foundation
+import SQLite3
+
+class MuxDB {
+    static let shared = MuxDB()
+    
+    let SQLITE_TRANSIENT = unsafeBitCast(-1, to: sqlite3_destructor_type.self)
+    
+    // specifies to store database in Files app on user's device
+    private var dbPath = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0] as String
+    
+    private var db: OpaquePointer?
+    
+    private init() {
+        db = openDatabase()
+        createTable()
+    }
+    // create database
+    func openDatabase() -> OpaquePointer? {
+        
+        let url = NSURL(fileURLWithPath: dbPath)
+        
+        // names db
+        if let pathComponent = url.appendingPathComponent("MuxDB") {
+            let filePath = pathComponent.path
+            if sqlite3_open(filePath, &db) == SQLITE_OK {
+                return db
+            }
+        }
+        
+        return db
+    }
+    
+    // create a table in database
+    
+    func createTable(){
+        
+        let createTable = sqlite3_exec(db, "CREATE TABLE IF NOT EXISTS Saved (Key TEXT NULL, Data TEXT NULL);", nil, nil, nil)
+        
+        if(createTable == SQLITE_OK) {
+            
+            let createIndex = sqlite3_exec(db, "CREATE UNIQUE INDEX IF NOT EXISTS key_domain ON Saved (Key);", nil, nil, nil)
```

Comment:
There's a compiler warning here that says `createIndex` is unused.

Generally there's no error handling in this code, which means if something went wrong the application won't know and will happily continue sending objects to this module. I'd say any call to `sqlite3_xxx()` should crash. You may want o define an internal function that takes a `sqlite3_xxx()` result and crashes if it's not `SQLITE_OK`, to simplify the code.